### PR TITLE
Swiftify API - Crashes

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [ ] ~Has `CHANGELOG.md` been updated?~
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests?~
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

These changes adapt the Crashes module API to swift-like style.

## Related PRs or issues

[AB#83317](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/83317)

### CODE REVIEW

File: AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h [hunk 1]
```diff
@@ -14,35 +14,27 @@ NS_ASSUME_NONNULL_BEGIN
...
 @interface MSACWrapperCrashesHelper : NSObject
 
 /**
- * Sets the crash handler setup delegate.
+ * Sets or gets the crash handler setup delegate.
  *
- * @param delegate The delegate to set.
  */
-+ (void)setCrashHandlerSetupDelegate:(id<MSACCrashHandlerSetupDelegate>)delegate;
+@property(class, nonatomic, setter=setCrashHandlerSetupDelegate:, getter=getCrashHandlerSetupDelegate) id<MSACCrashHandlerSetupDelegate>
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h:24]: Please read style guidelines for getters..

Wait, it was already there in this form.. anyway, let's do it properly

File: AppCenterCrashes/AppCenterCrashes/Model/MSACErrorReport.h [hunk 2]
```diff
@@ -63,6 +64,6 @@
  *
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashes/Model/MSACErrorReport.h:64]: Does `@return` work with properties? Please check generated docs

File: AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h [hunk 3]
```diff
@@ -14,35 +14,27 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  * This general class allows wrappers to supplement the Crashes SDK with their own behavior.
  */
+NS_SWIFT_NAME(WrapperCrashesHelper)
 @interface MSACWrapperCrashesHelper : NSObject
 
 /**
- * Sets the crash handler setup delegate.
+ * Sets or gets the crash handler setup delegate.
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h:21]: "Sets or gets " it's not needed for property

File: AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h [hunk 4]
```diff
@@ -14,35 +14,27 @@ NS_ASSUME_NONNULL_BEGIN
...
  *
- * @param delegate The delegate to set.
  */
-+ (void)setCrashHandlerSetupDelegate:(id<MSACCrashHandlerSetupDelegate>)delegate;
+@property(class, nonatomic, getter=getCrashHandlerSetupDelegate, setter=setCrashHandlerSetupDelegate:) id<MSACCrashHandlerSetupDelegate>
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h:24]: "get" violates the "Omit needless words" rule. 

See https://swift.org/documentation/api-design-guidelines/

File: AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.m [hunk 5]
```diff
@@ -33,6 +33,10 @@ + (void)setCrashHandlerSetupDelegate:(id<MSACCrashHandlerSetupDelegate>)delegate
   return [[MSACWrapperCrashesHelper sharedInstance] crashHandlerSetupDelegate];
 }
 
++ (id<MSACCrashHandlerSetupDelegate>)delegate {
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.m:36]: Why did you add/rename it?

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: AppCenterCrashes/AppCenterCrashes/MSACCrashes.h [hunk 6]
```diff
@@ -148,16 +149,15 @@ typedef NS_ENUM(NSUInteger, MSACUserConfirmation) {
...
  * @see MSACCrashesDelegate
  */
-+ (void)setDelegate:(_Nullable id<MSACCrashesDelegate>)delegate;
+@property(class, nonatomic, weak) id<MSACCrashesDelegate> _Nullable delegate;
 
 /**
  * Set a user confirmation handler that is invoked right before processing crash reports to determine whether sending crash reports or not.
  *
- * @param userConfirmationHandler A handler for user confirmation.
  *
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashes/MSACCrashes.h:157]:
```suggestion
```

File: AppCenterCrashes/AppCenterCrashes/MSACCrashes.h [hunk 7]
```diff
@@ -148,16 +149,15 @@ typedef NS_ENUM(NSUInteger, MSACUserConfirmation) {
...
  * @see MSACUserConfirmationHandler
  */
-+ (void)setUserConfirmationHandler:(_Nullable MSACUserConfirmationHandler)userConfirmationHandler;
+@property(class, nonatomic, weak) MSACUserConfirmationHandler _Nullable userConfirmationHandler;
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashes/MSACCrashes.h:160]:
```suggestion
@property(class, nonatomic) MSACUserConfirmationHandler _Nullable userConfirmationHandler;
```

File: AppCenterCrashes/AppCenterCrashes/MSACCrashes.h [hunk 8]
```diff
@@ -148,16 +149,15 @@ typedef NS_ENUM(NSUInteger, MSACUserConfirmation) {
  *
  * @see MSACCrashesDelegate
  */
-+ (void)setDelegate:(_Nullable id<MSACCrashesDelegate>)delegate;
+@property(class, nonatomic, weak) id<MSACCrashesDelegate> _Nullable delegate;
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashes/MSACCrashes.h:152]:
```suggestion
@property(class, nonatomic) id<MSACCrashesDelegate> _Nullable delegate;
```

File: AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h [hunk 9]
```diff
@@ -14,35 +14,27 @@ NS_ASSUME_NONNULL_BEGIN
...
 
 /**
- * Sets the crash handler setup delegate.
+ * The crash handler setup delegate.
  *
- * @param delegate The delegate to set.
  */
-+ (void)setCrashHandlerSetupDelegate:(id<MSACCrashHandlerSetupDelegate>)delegate;
+@property(class, nonatomic, weak, setter=setCrashHandlerSetupDelegate:) _Nullable id<MSACCrashHandlerSetupDelegate>
+сrashHandlerSetupDelegate;
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h:25]: Format. Setter name is default, no need to customize

File: AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h [hunk 10]
```diff
@@ -14,35 +14,27 @@ NS_ASSUME_NONNULL_BEGIN
...
-+ (id<MSACCrashHandlerSetupDelegate>)getCrashHandlerSetupDelegate;
-
-/**
- * Enables or disables automatic crash processing.
- *
- * @param automaticProcessing Passing NO causes SDK not to send reports immediately, even if "Always Send" is true.
+ * Enables or disables automatic crash processing. Passing NO causes SDK not to send reports immediately, even if "Always Send" is true.
  */
-+ (void)setAutomaticProcessing:(BOOL)automaticProcessing;
+@property(class, nonatomic, setter=setAutomaticProcessing:) BOOL automaticProcessing;
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h:30]: Same

File: AppCenterCrashes/AppCenterCrashes/MSACCrashes.h [hunk 11]
```diff
@@ -148,16 +149,14 @@ typedef NS_ENUM(NSUInteger, MSACUserConfirmation) {
...
 
 /**
  * Set a user confirmation handler that is invoked right before processing crash reports to determine whether sending crash reports or not.
  *
- * @param userConfirmationHandler A handler for user confirmation.
- *
  * @see MSACUserConfirmationHandler
  */
-+ (void)setUserConfirmationHandler:(_Nullable MSACUserConfirmationHandler)userConfirmationHandler;
+@property(class, nonatomic) MSACUserConfirmationHandler _Nullable userConfirmationHandler;
```
Comment (DmitriyKirakosyan) [inline AppCenterCrashes/AppCenterCrashes/MSACCrashes.h:159]: missing `setter= setUserConfirmationHandler`

File: AppCenterCrashes/AppCenterCrashes/Model/MSACErrorReport.h [hunk 12]
```diff
@@ -59,10 +60,8 @@
  * thread for too long, or other reasons. See Apple documentation:
  * https://developer.apple.com/library/ios/qa/qa1693/_index.html.
  *
- * @return YES if the details represent an app kill instead of a crash.
- *
  * @see `[MSACCrashes didReceiveMemoryWarningInLastSession]`
  */
-- (BOOL)isAppKill;
+@property(nonatomic, readonly) BOOL isAppKill;
```
Comment (DmitriyKirakosyan) [inline AppCenterCrashes/AppCenterCrashes/Model/MSACErrorReport.h:65]: should be `NS_SWIFT_NAME(appKill)`

File: AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h [hunk 13]
```diff
@@ -14,35 +14,26 @@ NS_ASSUME_NONNULL_BEGIN
...
-+ (id<MSACCrashHandlerSetupDelegate>)getCrashHandlerSetupDelegate;
-
-/**
- * Enables or disables automatic crash processing.
- *
- * @param automaticProcessing Passing NO causes SDK not to send reports immediately, even if "Always Send" is true.
+ * Enables or disables automatic crash processing. Passing NO causes SDK not to send reports immediately, even if "Always Send" is true.
  */
-+ (void)setAutomaticProcessing:(BOOL)automaticProcessing;
+@property(class, nonatomic) BOOL automaticProcessing;
```
Comment (DmitriyKirakosyan) [inline AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h:29]: missing `setter`

File: AppCenterCrashes/AppCenterCrashes/MSACCrashes.h (hunk 11 above)
Comment (AnastasiaKubova) [reply AppCenterCrashes/AppCenterCrashes/MSACCrashes.h:159]: https://github.com/microsoft/appcenter-sdk-apple/pull/2194#discussion_r507561854

File: AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h (hunk 13 above)
Comment (AnastasiaKubova) [reply AppCenterCrashes/AppCenterCrashes/WrapperSDKUtilities/MSACWrapperCrashesHelper.h:29]: https://github.com/microsoft/appcenter-sdk-apple/pull/2194#discussion_r507562523

**Review by MatkovIvan — APPROVED** [review]
Merging this to the common feature branch, future edits will be done separately
